action.yml

```diff
@@ -34,53 +34,74 @@ runs:
       shell: bash
       env:
         TYPE: ${{ steps.gettype.outputs.type }}
+        GITHUB_REPOSITORY: ${{ github.repository }}
       run: |
         major_type=$(echo $TYPE | cut -d "," -f 1)
         minor_type=$(echo $TYPE | cut -d "," -f 2)
-        # https://docs.github.com/en/rest/branches/branches#list-branches
         RESP_CODE=$(curl -w %{http_code} -s -L -o __branches.json \
           -H "Accept: application/vnd.github+json" \
-          -H "Authorization: Bearer ${{ github.token }}"\
+          -H "Authorization: Bearer ${{ github.token }}" \
           -H "X-GitHub-Api-Version: 2022-11-28" \
           https://api.github.com/repos/$GITHUB_REPOSITORY/branches \
         )
         if [[ $RESP_CODE != "200" ]]; then
           echo "Failed to list branches - HTTP response code was $RESP_CODE"
           exit 1
         fi
+        RESP_CODE=$(curl -w %{http_code} -s -L -o __tags.json \
+          -H "Accept: application/vnd.github+json" \
+          -H "Authorization: Bearer ${{ github.token }}" \
+          -H "X-GitHub-Api-Version: 2022-11-28" \
+          https://api.github.com/repos/$GITHUB_REPOSITORY/tags \
+        )
+        if [[ $RESP_CODE != "200" ]]; then
+          echo "Failed to list branches - HTTP response code was $RESP_CODE"
+          exit 1
+        fi
         major_branch=""
         minor_branch=""
         major_branches=$(jq -r '.[] | select(.name | test("^[0-9]+$")) | .name' __branches.json | sort -V -r)
         # "major_branches" needs to be quoted to preserve line-breaks
         major_branches_count=$(echo "$major_branches" | wc -l)
-        if [[ $major_type == "latest" ]]; then
-            if (( $major_branches_count < 1 )); then
-                echo "No major branches found"
-                exit 1
-            fi
-            major_branch=$(echo "$major_branches" | head -1)
-        elif [[ $major_type == "previous" ]]; then
-            if (( $major_branches_count < 2 )); then
-                echo "Only one major branch found, not running CI on this cycle - will run on next cycle"
-                exit 0
-            fi
+        if (( $major_branches_count < 1 )); then
+          echo "No major branches found, cannot continue"
+          exit 1
+        fi
+        latest_major_branch=$(echo "$major_branches" | head -1)
+        latest_major_tags_count=0
+        # not escaping rx . as \. because jq will say it is a compile error
+        latest_major_tags=$(jq -r ".[] | select(.name | test(\"^$latest_major_branch.[0-9]+.[0-9]+$\")) | .name" __tags.json | sort -V -r)
```

Comment:
Can you please either add the word `stable` to the variable name or add a comment saying we explicitly want the stable tag?

Reply:
Updated